flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/config/AWSConfigConstants.java

```diff
@@ -154,6 +154,77 @@ public enum CredentialProvider {
     /** Read Request timeout for {@link SdkAsyncHttpClient}. */
     public static final String HTTP_CLIENT_READ_TIMEOUT_MILLIS = "aws.http-client.read-timeout";
 
+    /**
+     * The type of {@link software.amazon.awssdk.http.SdkHttpClient}. If set, all AWS clients will
+     * use this specified HTTP client. If not set, HTTP_CLIENT_TYPE_DEFAULT will be used. For
+     * specific types supported, see HTTP_CLIENT_TYPE_* defined below.
+     */
+    public static final String HTTP_CLIENT_TYPE = "http-client.type";
+
+    // ---- glue configs
+
+    /**
+     * Used to configure the connection timeout in milliseconds for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see <a
+     * href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html">...</a>
+     */
+    public static final String HTTP_CLIENT_CONNECTION_TIMEOUT_MS =
+            "http-client.connection-timeout-ms";
```

Comment:
hmmm, do you think we need to make it consistent with existing http-client configs like `HTTP_CLIENT_READ_TIMEOUT_MILLIS` where prefix is `aws.http-cllient` I am aware this might need some rework of OptionsUtils implementations. Wdyt? can loop in Danny for a second opinion.

Reply:
bump